developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -107,4 +111,72 @@ export class Compiler {
       return false;
     }
   }
-}
\ No newline at end of file
+
+  /**
+   *
+   * @param pattern UnicodeSet pattern such as `[a-z]`
+   * @param bufferSize guess as to the buffer size
+   * @returns UnicodeSet accessor object
+   */
+  public async parseUnicodeSet(pattern: string, bufferSize: number) : Promise<UnicodeSet> {
+    if (!bufferSize) {
+      bufferSize = 100;
+    }
+
+    const initOk = await this.init();
+    if (!initOk) {
+      throw Error(`WASM machinery didn't start up properly`);
+    }
```

Comment:
init should be done separately to keep the async out of this func; also we have `reportMessage` rather than `throw` for handling errors? Perhaps we need a design discussion on error handling throughout kmc as it is currently expanding rapidly.

```suggestion
    if(!this.wasmModule) {
      this.callbacks.reportMessage(CompilerMessages.Fatal_MissingWasmModule());
      return null;
    }
```

Reply:
that might be good,  failed ot startup would definitely be fatal.

Reply:
done